src/Libraries/LG/Source/lg.h

```diff
@@ -37,7 +37,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "log.h"
 
 // Relative path workaround
-#define THIS_FILE ((strrchr(__FILE__, '/') ?: __FILE__ - 1) + 1)
+#define THIS_FILE ((strrchr(__FILE__, '/') ? 0 : __FILE__ - 1) + 1)
```

Comment:
Another GNU extension. An empty first branch in a ?: expression evaluates to the condition, not null. What the macro is doing is taking the leaf from the filename; but is it really necessary?

Reply:
MSVC flatout refuses to compile the ternary with a blank true block.

Reply:
Yes, it's a GNU extension.
The portable way to achieve the same effect would be to do
#define THIS_FILE (strrchr(__FILE__, '/') ? strrchr(__FILE__, '/') + 1 : __FILE__)
But that would evaluate the strrchr() twice if the result were non-null.
The alternative would be not to bother with the THIS_FILE macro at all but just to display the filename in all its hierarchical glory. That's pretty ugly though. Probably best to just suck up the extra evaluation. I'm sure we're not spending that much time formatting debug messages, relatively speaking.